lib/job.js

```diff
@@ -157,6 +163,30 @@ class Job extends BaseModel {
             .then(successfulBuilds => successfulBuilds[0]);
     }
 
+    /**
+     * Update a job
+     * @method update
+     * @return {Promise}
+     */
+    update() {
+        // Lazy load factory dependency to prevent circular dependency issues
+        // https://nodejs.org/api/modules.html#modules_cycles
+        /* eslint-disable global-require */
+        const PipelineFactory = require('./pipelineFactory');
+        /* eslint-enable global-require */
+        const pipelineFactory = PipelineFactory.getInstance();
+
+        return super.update()
+            // Update for periodic job
+            .then(newJob => pipelineFactory.get(newJob.pipelineId)
+                .then(pipeline => this[executor].startPeriodic({
+                    pipeline,
+                    job: newJob,
+                    tokenGen: this[tokenGen],
+                    isUpdate: true
+                })));
```

Comment:
We should still return the job object in the end for the update function, that's the expected behavior